packages/libs/react-ui/src/components/Accordion/Accordion.tsx

```diff
@@ -1,43 +1,54 @@
-'use client';
+import { IAccordionSectionProps } from '.';
 
-import { AccordionSection, IAccordionSectionProps } from './AccordionSection';
+import type { FC, FunctionComponentElement } from 'react';
+import React, { useEffect, useState } from 'react';
 
-import React, { FC, useState } from 'react';
-
-export interface IAccordionProps {
-  sections: Omit<IAccordionSectionProps, 'isOpen' | 'onToggle'>[];
+export interface IAccordionRootProps {
+  children?: FunctionComponentElement<IAccordionSectionProps>[];
   linked?: boolean;
+  openSection?: number;
 }
 
-export const Accordion: FC<IAccordionProps> = ({ sections, linked = true }) => {
-  const [expandedSections, setExpandedSections] = useState<number[]>([]);
-  const handleOpen = (index: number): void => {
-    if (linked) setExpandedSections([index]);
-    else setExpandedSections([...expandedSections, index]);
-  };
-  const handleClose = (index: number): void => {
-    setExpandedSections(expandedSections.filter((item) => item !== index));
-  };
-
-  const isOpen = (index: number): boolean => expandedSections.includes(index);
+export const AccordionRoot: FC<IAccordionRootProps> = ({
+  children,
+  linked = false,
+  openSection = -1,
+}) => {
+  const [openSections, setOpenSections] = useState([openSection]);
 
-  const handleToggle = (index: number): void => {
-    if (isOpen(index)) handleClose(index);
-    else handleOpen(index);
-  };
+  useEffect(() => {
+    if (linked && openSections.length > 1) {
+      const lastOpen = openSections.pop() || -1;
+      setOpenSections([lastOpen]);
+    }
+  }, [linked]);
 
   return (
-    <div data-testid="kda-accordion-wrapper">
-      {sections.map((section, index) => (
-        <AccordionSection
-          {...section}
-          isOpen={isOpen(index)}
-          onToggle={() => handleToggle(index)}
-          key={String(section.title)}
-        >
-          {section.children}
-        </AccordionSection>
-      ))}
+    <div data-testid="kda-accordion-sections">
+      {React.Children.map(children, (section, sectionIndex) =>
+        React.cloneElement(
+          section as React.ReactElement<
+            HTMLElement | IAccordionSectionProps,
+            | string
+            | React.JSXElementConstructor<JSX.Element & IAccordionSectionProps>
+          >,
+          {
+            index: sectionIndex,
+            openSections,
+            onClick: () => {
+              if (openSections.includes(sectionIndex)) {
+                setOpenSections(openSections.filter((i) => i !== sectionIndex));
+                section?.props.onClose?.();
+              } else {
+                setOpenSections(
+                  linked ? [sectionIndex] : [...openSections, sectionIndex],
+                );
+                section?.props.onOpen?.();
+              }
```

Comment:
I already think this is a huge improvement! Some more comments, I think it's better to keep this as simple as possible so I have the following suggestions: 
- Move the `onOpen` and `onClose` function calls to the `Accordion.Section` component. The component should already be aware of whether it is open, so I think it makes sense to have an onClick handler inside of the component that handles function calls that are only associated with that component. I feel like this would be more clear since `Accordion.Root` has no information about the section props
- Nit: This isn't super necessary, but could be nice to make a higher-order function that returns the onClick handler for these sections to keep the jsx cleaner